src/internal/functions/Add-DscResourceModule.ps1

```diff
@@ -0,0 +1,60 @@
+function Add-DscResourceModule {
+  <#
+  .SYNOPSIS
+    Vendor PowerShell modules into the build folder either by path or by retrieving the modules from the Gallery
+
+  .DESCRIPTION
+    Given a DSC Resource Info object, introspect on the source (if possible) to return information about
+    the resources parameters, including their help text, whether or not they are mandatory for calling the
+    resource with the Get method, and whether or not they are mandatory for calling the resource with the
+    Set method.
+
+    This information can be reliably retrieved if the DSC resource is implemented in PowerShell and is
+    not class based. Otherewise, best guesses are made about the mandatory status of the parameters via
+    the Required attribute and the help information is null.
+
+  .PARAMETER Name
+    Specifies the name of the module to save.
+
+  .PARAMETER Path
+    Specifies the location on the local computer to store the saved module.
+
+  .PARAMETER RequiredVersion
+    Specifies the exact version number of the module to save.
+    If left blank, will default to latest available.
+
+  .EXAMPLE
+    Add-DscResourceModule -TargetDir ./tmp -Name PowerShellGet -Version 2.2.3
```

Comment:
When exporting the function, the pester tests will complain that there's not an explainer of the example underneath it - can be as simple as:

```Suggestion
Add-DscResourceModule -TargetDir ./tmp -Name PowerShellGet -Version 2.2.3

This example will search the PowerShell gallery for version `2.2.3` of the PowerShellGet and,
if it finds it, save the module and its dependencies into `./tmp`.
```

Reply:
added this example